CMakeLists.txt

```diff
@@ -91,6 +91,23 @@ if (ENABLE_HARDENING)
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${HARDENING_LDFLAGS}")
 endif()
 
+option(DEV_BUILD "Enable additional build options to simplify debugging" OFF)
+
+# Assuming the C and CXX compiler IDs are equal.
+# For clang `MATCHES` is used to match both Clang and AppleClang.
+if (CMAKE_C_COMPILER_ID MATCHES "Clang" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL "10" OR
+    CMAKE_C_COMPILER_ID STREQUAL "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL "8")
+    add_compile_flags("C;CXX" "-fmacro-prefix-map=${CMAKE_SOURCE_DIR}=.")
+endif()
+
+# Assuming the C and CXX compiler IDs are equal.
+# For clang `MATCHES` is used to match both Clang and AppleClang.
+if (NOT DEV_BUILD AND
```

Comment:
This condition is FALSE when DEV_BUILD is FALSE and GNU >= 4.4.7, probably parentheses after AND are missed.

Reply:
Turned out that `AND` and `OR` logical operators have the same [precedence](https://cmake.org/cmake/help/latest/command/if.html#condition-syntax)...

> Binary logical operators [AND](https://cmake.org/cmake/help/latest/command/if.html#and) and [OR](https://cmake.org/cmake/help/latest/command/if.html#or), from left to right, without any short-circuit.